lib/actions/addClass.js

```diff
@@ -0,0 +1,30 @@
+class Add {
+  constructor(){
+    this.success = false;
+  }
+
+  validator(){}
+
+  async execute(arg, project, db = {store(){}}){ 
```

Comment:
We want this execute method to do basically 2 things. 

1. Store the link data somewhere (using the `db` parameter to accomplish that. we can fake it for now)
2. Return a promise that will, in-turn return the structured link data.

Right now, this is doing number one, but not number 2.

Reply:
does this mean that I want my db.store() method to return that current structured link data?

Reply:
Not necessarily, no. Think of @nrambeck's comments as 2 actions happening in this method. Save this data to a table record. Then return an object with structured data. Those are 2 different steps in this method.

We haven't done any of the work to actually setup anything regarding a database or call into one. For that reason, it's reasonable to _NOT_ do the first bullet. 

Instead, let's return a new `Response` instance with some data that is necessary to compose a response. The following feels like a good structure (it's not a class of course, but it should give you an idea of the "shape" of a `Response` class:

```
{ 
  label,
  link,
  project
}
```

The `Response` class that holds :point_up: that data, would then (per our diagram and such), have that `sendViaBot` method to turn this into a slack response :boom:!